src/Cli/func/Helpers/PythonHelpers.cs

```diff
@@ -391,37 +442,111 @@ public static async Task<Stream> ZipToSquashfsStream(Stream stream)
         {
-            var md5File = Path.Combine(pythonPackages, $"{Constants.RequirementsTxt}.md5");
+            var md5FileName = GetDependencyMd5FileName(packageTool);
+            var md5File = Path.Combine(pythonPackages, md5FileName);
             if (!FileSystemHelpers.FileExists(md5File))
             {
                 return false;
             }
 
             var packagesMd5 = await FileSystemHelpers.ReadAllTextFromFileAsync(md5File);
-            var requirementsTxtMd5 = SecurityHelpers.CalculateMd5(requirementsTxt);
+            var currentMd5 = CalculateDependencyChecksum(functionAppRoot, packageTool);
+
+            return packagesMd5 == currentMd5;
+        }
+
+        private static string[] GetDependencyFileNames(PythonPackageTool packageTool)
+        {
+            return packageTool switch
+            {
+                PythonPackageTool.Pip => new[] { Constants.RequirementsTxt },
+                PythonPackageTool.Poetry => new[] { Constants.PyProjectToml },
+                PythonPackageTool.Uv => new[] { Constants.PyProjectToml, Constants.UvLock },
+                _ => Array.Empty<string>()
+            };
+        }
+
+        private static string GetDependencyMd5FileName(PythonPackageTool packageTool)
+        {
+            return packageTool switch
+            {
+                PythonPackageTool.Pip => $"{Constants.RequirementsTxt}.md5",
+                PythonPackageTool.Poetry => $"{Constants.PyProjectToml}.md5",
+                PythonPackageTool.Uv => Constants.UvMd5FileName,
+                _ => "dependencies.md5"
+            };
+        }
+
+        private static string[] GetDependencyFilePaths(string functionAppRoot, PythonPackageTool packageTool)
+        {
+            return packageTool switch
+            {
+                PythonPackageTool.Pip => new[] { Path.Combine(functionAppRoot, Constants.RequirementsTxt) },
+                PythonPackageTool.Poetry => new[] { Path.Combine(functionAppRoot, Constants.PyProjectToml) },
+                PythonPackageTool.Uv => new[] 
+                { 
+                    Path.Combine(functionAppRoot, Constants.PyProjectToml),
+                    Path.Combine(functionAppRoot, Constants.UvLock)
+                },
+                _ => Array.Empty<string>()
+            };
+        }
 
-            return packagesMd5 == requirementsTxtMd5;
+        private static string CalculateDependencyChecksum(string functionAppRoot, PythonPackageTool packageTool)
+        {
+            switch (packageTool)
+            {
+                case PythonPackageTool.Pip:
+                    var reqTxtFile = Path.Combine(functionAppRoot, Constants.RequirementsTxt);
+                    return SecurityHelpers.CalculateMd5(reqTxtFile);
+
+                case PythonPackageTool.Poetry:
+                    var pyProjectFile = Path.Combine(functionAppRoot, Constants.PyProjectToml);
```

Comment:
Move this variable outside the `switch` block since its being used in both Poetry and Uv. This will make the code more cleaner